src/xbot/common/logic/Latch.java

```diff
@@ -22,33 +22,32 @@ public Latch(boolean initialValue, EdgeType latchType) {
     public void setValue(boolean value) {
         boolean different = this.value != value;
 
-        switch (latchType) {
-        case Both:
-            if (different) {
-                saveValueAndAlertWatchers(value);
+        if (different) {
+            switch (latchType) {
+                case Both:
+                    saveValueAndAlertWatchers(value);
+                    break;
+                case RisingEdge:
+                    if (value) {
+                        saveValueAndAlertWatchers(true);
+                    }
+                    break;
+                case FallingEdge:
+                    if (!value) {
+                        saveValueAndAlertWatchers(false);
+                    }
+                    break;
+                default:
+                    this.value = value;
+                    break;
             }
-            break;
-        case RisingEdge:
-            if (different && value) {
-                saveValueAndAlertWatchers(value);
-            }
-            break;
-        case FallingEdge:
-            if (different && !value) {
-                saveValueAndAlertWatchers(value);
-            }
-            break;
-        default:
-            break;
         }
     }
 
     private void saveValueAndAlertWatchers(boolean value) {
-        this.value = value;
```

Comment:
What am I missing -- isn't this where we saved the previous value?

Reply:
oops some how that passed the test